modules/lib/src/main/resources/assets/js/app/ui2/dialog/ConfirmationDialog.tsx

```diff
@@ -0,0 +1,133 @@
+import {Button, Dialog, cn} from '@enonic/ui';
+import type {ComponentPropsWithoutRef, ReactElement, ReactNode} from 'react';
+import {createContext, useCallback, useContext, useMemo, useState} from 'react';
+import {i18n} from '@enonic/lib-admin-ui/util/Messages';
+import {forwardRef} from 'react';
+
+//
+// Confirmation dialog context
+//
+
+type ConfirmationDialogContextValue = {
+    unlocked: boolean;
```

Comment:
Now I think it would be better to make it more clear: `unlocked` -> `submitEnabled`. We'll need to rename all other related props too.